### ISSUE TITLE
[MATLAB] Consider lowering the minimum CMake version requirement for building the MATLAB interface

### ISSUE THREAD

**Issue by kevingurney** [apache/arrow#35221]
### Describe the enhancement requested

As discussed in this [comment thread](https://github.com/apache/arrow/pull/34563#discussion_r1137882276), the MATLAB interface currently defines a relatively high minimum CMake version (i.e. [`cmake_minimum_required`](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html)) of `3.20`. For comparison, [the C++ libraries define a minimum required CMake version of `3.5`](https://github.com/apache/arrow/blob/72d57ac4ccc1c80003481d04ad23159c5ed2c2f1/cpp/CMakeLists.txt#L18).

The minimum CMake version for the MATLAB interface was originally [increased from `3.2` to `3.20` to work around some bugs in `FindMatlab` that were making development more challenging](https://github.com/apache/arrow/pull/10305#discussion_r657176867).

It may be worth revisiting this decision and doing a thorough investigation into whether there are any opportunities to lower the minimum required CMake version to ensure broader compatibility with older CMake versions.

This would be a good first issue for any community members who are interested in contributing to the MATLAB interface and would like to get some experience using CMake. This would involve building the interface against versions of CMake prior to `3.20` on Windows, macOS, and Linux, and seeing whether there are any issues that come up.

### Component(s)

MATLAB

**Comment by kevingurney** [comment]
I don't believe I have the permissions to add additional labels to this issue. However, I think it would make sense to add the [`good-first-issue`](https://github.com/apache/arrow/labels/good-first-issue) and/or [`good-second-issue`](https://github.com/apache/arrow/labels/good-second-issue) labels to this issue.

**Comment by kou** [comment]
I've added them.

FYI: The C++ implementation requires CMake 3.16 or later soon: #34921

**Comment by kevingurney** [comment]
Thanks for adding the labels and for pointing this out @kou!

Given this information, it probably wouldn't make sense to go any lower than `3.16` for the minimum required CMake version for the MATLAB interface since it depends on the `CMakeLists.txt` for the the C++ libraries.

**Comment by thisisnic** [comment]
This issue has been marked as stale because it has had no activity in the past 2 years. Please remove the stale label or comment below, or this issue will be closed in 14 days.
